src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -17,42 +19,61 @@ object SignedTransaction {
   val LastByteOfAddress: Int = FirstByteOfAddress + AddressLength
   val negativePointSign = 27
   val positivePointSign = 28
-  val allowedSignValues: Set[Int] = Set(negativePointSign, positivePointSign)
-
 }
 
 case class SignedTransaction(
   tx: Transaction,
-  pointSign: Byte, //v - 27 or 28 according to yellow paper, but it is 37 and 38 in ETH
+  pointSign: Byte, //v
   signatureRandom: ByteString, //r
   signature: ByteString /*s*/) {
 
   import SignedTransaction._
 
-  lazy val bytesToSign: Array[Byte] =
+  lazy val bytesToSign: Array[Byte] = if (pointSign == negativePointSign || pointSign == positivePointSign) {
+    //global transaction
+    crypto.sha3(
+      rlpEncode(RLPList(
+        tx.nonce,
+        tx.gasPrice,
+        tx.gasLimit,
+        tx.receivingAddress.toArray,
+        tx.value,
+        tx.payload)))
+  } else {
+    //chain specific transaction
     crypto.sha3(
       rlpEncode(RLPList(
-                  tx.nonce,
-                  tx.gasPrice,
-                  tx.gasLimit,
-                  tx.receivingAddress.toArray,
-                  tx.value,
-                  tx.payload)))
+        tx.nonce,
+        tx.gasPrice,
+        tx.gasLimit,
+        tx.receivingAddress.toArray,
+        tx.value,
+        tx.payload,
+        Config.Blockchain.chainId,
+        0,
+        0)))
+  }
 
 
   lazy val recoveredAddress: Option[Array[Byte]] =
     recoveredPublicKey.map(key => crypto.sha3(key).slice(FirstByteOfAddress, LastByteOfAddress))
 
-  //todo implement EC-72
-  lazy val recoveredPublicKey: Option[Array[Byte]] = if (allowedSignValues.contains(pointSign)) {
-    ECDSASignature.recoverPubBytes(
-      new BigInteger(1, signatureRandom.toArray[Byte]),
-      new BigInteger(1, signature.toArray[Byte]),
-      ECDSASignature.recIdFromSignatureV(pointSign),
-      bytesToSign
-    )
-  } else {
-    None
+  lazy val recoveredPublicKey: Option[Array[Byte]] = pointSign match {
+    case p if p == negativePointSign || p == (Blockchain.chainId * 2 + 35).toByte =>
+      ECDSASignature.recoverPubBytes(
+        new BigInteger(1, signatureRandom.toArray[Byte]),
+        new BigInteger(1, signature.toArray[Byte]),
+        ECDSASignature.recIdFromSignatureV(negativePointSign),
+        bytesToSign
+      )
+    case p if p == positivePointSign || p == (Blockchain.chainId * 2 + 36).toByte =>
```

Comment:
Somehow scalastyle doesn't qualify that as a magic number. I still think it would be nice to have those numbers as vals and/or provide comments where they came from.